### DISCUSSION TITLE
New user

### DISCUSSION THREAD

**Discussion by Joy1994j** [community/community#135342] [New to GitHub]
i am new please help me

**Comment by tbrowne5** [comment]
GitHub is a web-based platform used for version control and collaborative software development. It uses Git, a version control system that tracks changes to your files, so you can revert back to previous versions or collaborate with others on projects.

1. Install Git -- this is a command-line tool that you need to install on your computer to work with GitHub repositories. Visit https://git-scm.com/ to install

2. Configure Git -- After installation, open your terminal, set up your username and email like so...

```
git config --global user.name "Your Name"
git config --global user.email "your.email@example.com"
```
3. Move on to trying things out like cloning a repository (aka repo):

`git clone https://github.com/username/repository.git`

4. You can also create a new repo on GitHub from the website by clicking on the "new" button on the GitHub dashboard. Give it a name and decide whether it should be public or private. After creating a repo, you can add files to it.

5. Once you've added files to a repo, or made changed, you can add those changes, commit them, and push them to github:

```
git add .
git commit -m "Initial commit"
git push origin main
```

5. Branches -- branches allow you to work on different parts of a project separately. To create a new branch:

`git checkout -b new-branch-name`

5b. To switch between branches:

`git checkout branch-name`

5c. Merging Branches -- once you’ve finished working on a branch, you can merge it back into the main branch:
```
git checkout main
git merge new-branch-name
```

This guide should help you get started with GitHub, but there's also a ton of tutorials out there to check out. Good luck!

**Comment by ghostinhershell** [comment]
Welcome to GitHub and the GitHub Community @Joy1994j 🎆 ! We're glad you're here and can't wait to see where your GitHub journey takes you. Check out this [check-in](https://github.com/orgs/community/discussions/42302#discussion-4683140) with links on how to choose your first project to contribute to, a blog with getting started resources, and a new user checklist.

Best of luck!
